lib/Maintenance/Tasks/LogArchiveTask.php

```diff
@@ -79,5 +89,28 @@ public function execute()
             $db->query('INSERT INTO '.$tablename.' '.sprintf($sql, '*'));
             $db->query('DELETE FROM '.ApplicationLoggerDb::TABLE_NAME.' WHERE `timestamp` < DATE_SUB(FROM_UNIXTIME('.$timestamp.'), INTERVAL '.$archive_treshold.' DAY);');
         }
+
+        $lockId = self::class;
+        if (!Lock::isLocked($lockId, 86400) && date('H') <= 4) {
+            // execution should be only sometime between 0:00 and 4:59 -> less load expected
+            Lock::lock($lockId);
+            $this->logger->debug('Deleting referenced FileObjects of application_logs which are older than '. $archive_treshold.' days');
+            $fileIterator = new \DirectoryIterator(PIMCORE_LOG_FILEOBJECT_DIRECTORY);
+            $fileIterator = new \CallbackFilterIterator(
+                $fileIterator,
+                static function (\SplFileInfo $fileInfo) use ($archive_treshold) {
+                    return time() - $archive_treshold * 86400 > $fileInfo->getMTime();
```

Comment:
`time() - $archive_treshold * 86400` should be calculated outside the closure. There's no need to repeat it every time.

Reply:
You are right, fixed that. Thanks.